### PR TITLE
Deprecate terrain IsWater check - Part 1

### PR DESCRIPTION
**Pull request by reaperrr**
This removes all remaining `IsWater` checks except for `CreateEffectWarhead`. The latter is better done in its own PR, because it would make sense to refactor `ImpactTypes` along with it, which might require some rather large upgrade rule and amount of yaml changes.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/WithCrateBody.cs [hunk 1]
```diff
@@ -74,7 +79,9 @@ void INotifyParachuteLanded.OnLanded(Actor ignore)
 
 		void PlaySequence()
 		{
-			var sequence = self.World.Map.GetTerrainInfo(self.Location).IsWater ? info.WaterSequence : info.LandSequence;
+			var cell = self.Location;
+			onWater = self.World.Map.Contains(cell) && info.WaterTerrainTypes.Contains(self.World.Map.GetTerrainInfo(cell).Type);
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/WithCrateBody.cs]: Just curious: Why did you add `self.World.Map.Contains(cell)` here?

**Comment by Mailaender** [comment]
:+1:

**Comment by pchote** [comment]
Yay!  Thanks for making some progress on this.

File: OpenRA.Mods.Common/Traits/Render/WithCrateBody.cs (hunk 1 above)
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Render/WithCrateBody.cs]: Originally I wanted to make sure the sequence is only played when the cell is in the visible map area. But to be honest, I'm not sure such a safeguard is really needed, and I see now that this would be the wrong place to put it anyway, as it would just switch to LandSequence instead. I guess I'll just remove it.

**Comment by abcdefg30** [comment]
Is there a specific reason why you used `bool onWater` and not just `var onWater`?

**Comment by Phrohdoh** [comment]
`var` cannot be used outside of a method body.

**Comment by abcdefg30** [comment]
> `var` cannot be used outside of a method body.

I meant to implicitly ask why we need this variable outside of the method body.

File: OpenRA.Mods.Common/Traits/Parachutable.cs [hunk 2]
```diff
@@ -47,6 +51,8 @@ class Parachutable : INotifyParachuteLanded
 		readonly ParachutableInfo info;
 		readonly IPositionable positionable;
 
+		bool onWater;
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Parachutable.cs]: abcdefg30 is right here, this doesn't need to be global. Same in the other file below.

**Comment by obrakmann** [comment]
Looks fine otherwise, :+1:

**Comment by reaperrr** [comment]
Updated.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8b66ea95d7e16c270ccbf9b23fee73fed56e52b6)
